src/Agent/NewRelic/Agent/Core/Transformers/TransactionTransformer/ErrorEventMaker.cs

```diff
@@ -44,13 +57,40 @@ public ErrorEventWireModel GetErrorEvent(ImmutableTransaction immutableTransacti
             var priority = immutableTransaction.Priority;
 
             _attribDefs.GetTypeAttribute(TypeAttributeValue.TransactionError).TrySetDefault(attribValues);
-            SetErrorGroup(immutableTransaction.TransactionMetadata.ReadOnlyTransactionErrorState.ErrorData, attribValues);
+            SetErrorGroup(immutableTransaction.TransactionMetadata?.ReadOnlyTransactionErrorState?.ErrorData, attribValues);
             return new ErrorEventWireModel(attribValues, isSynthetics, priority);
         }
 
+        private IList<string> GetFormattedStackTrace(ErrorData errorData)
+        {
+            if (errorData.StackTrace == null)
+            {
+                return null;
+            }
+
+            var stackTrace = StackTraces.ScrubAndTruncate(errorData.StackTrace, _configurationService.Configuration.StackTraceMaximumFrames);
+            return stackTrace;
+        }
+
         private void SetErrorGroup(ErrorData errorData, IAttributeValueCollection attribValues)
         {
-            _attribDefs.ErrorGroup.TrySetValue(attribValues, errorData.ErrorGroup);
+            if (_configurationService.Configuration.ErrorGroupCallback == null)
+            {
+                return;
+            }
+
+            using (_agentTimerService.StartNew(SetErrorGroupSupportabilityName))
```

Comment:
I was thinking we'd put the timer around just the callback invocation. But maybe the little bit of extra work occurring here isn't enough to matter.

Reply:
I can see value in both. There could be a potential performance issue around generating the dictionary, but timing just the callback may be a clearer "this problem is definitely on your end" signal.

Reply:
I waffled back and forth between the two and went with timing most of the work, but if we want to just time the callback, I am totally fine with that.

Reply:
Updated to time callback only.